From 925a4c15cc7aca3c7c8ed970de23eed6b1a2ae65 Mon Sep 17 00:00:00 2001
From: Peter Eisenmann <p3732@getgoogleoff.me>
Date: Mon, 1 Apr 2024 00:57:14 +0200
Subject: [PATCH 1/3] audio-video-properties: return early without discoverer

Without a correctly initialized GstDiscoverer, no locations can be
set, so return early if that is the case.
---
 extensions/audio-video-properties/totem-properties-view.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/extensions/audio-video-properties/totem-properties-view.c b/extensions/audio-video-properties/totem-properties-view.c
index 2adf97cd1a..51507fdf97 100644
--- a/extensions/audio-video-properties/totem-properties-view.c
+++ b/extensions/audio-video-properties/totem-properties-view.c
@@ -441,12 +441,14 @@ totem_properties_view_set_location (TotemPropertiesView *props,
 {
     g_assert (TOTEM_IS_PROPERTIES_VIEW (props));
 
-    if (props->priv->disco)
+    if (props->priv->disco == NULL)
     {
-        gst_discoverer_stop (props->priv->disco);
+        return;
     }
 
-    if (location != NULL && props->priv->disco != NULL)
+    gst_discoverer_stop (props->priv->disco);
+
+    if (location != NULL)
     {
         gst_discoverer_start (props->priv->disco);
 
-- 
GitLab
